File: notebook/demo/src/test4.cpp

#include "cregion4.h"
#include <iostream>
using namespace std;

//  T,p
static const double tab35[3][2] = {
    {300, 0.00353658941},
    {500, 2.63889776},
    {600, 12.3443146}};

// p,T
static const double tab36[3][2] = {
    {0.1, 372.755919},
    {1, 453.035632},
    {10, 584.149488}};

void testSaturationline(void)
{
  //  Saturation line

  for (int i = 0; i < 3; i++)
  {
    double T = tab35[i][0];
    cout<<"("<<T<<","<<pSat(T)<<")"<<endl;
  }

  for (int i = 0; i < 3; i++)
  {
    double p = tab36[i][0];
    TSat(p);
    cout<<"("<<p<<","<<TSat(p)<<")"<<endl;
  }
}

int main(void)
{
  testSaturationline();
}

File: notebook/demo/src/test4.py

import unittest
from region4 import *

class Region4Test (unittest.TestCase):

    def setUp(self):
        # IF97-dev,Table35 Page 34 : T(K) p(MPa)
        self.tab35=[[300, 0.353658941e-2],
                    [500, 0.263889776e1],
                    [600, 0.123443146e2]]

        # IF97-dev, Table 36 Page 36 :  p(MPa) T(K)
        self.tab36=[[0.1, 0.372755919e3],
                    [  1, 0.453035632e3],
                    [ 10, 0.584149488e3]]

    def test_pSat(self):
        places = 6
        for item in  self.tab35:
             self.assertAlmostEqual(pSat(item[0]),item[1],places)

    def test_TSat(self):
        places = 6
        for item in  self.tab36:
             self.assertAlmostEqual(TSat(item[0]),item[1],places)
     

if __name__ == '__main__':
    unittest.main()       
